src/agent/reqwest-retry/src/lib.rs

```diff
@@ -109,8 +113,8 @@ mod test {
             .await;
 
         if let Err(err) = &resp {
-            let as_text = format!("{}", err);
-            assert!(as_text.contains("request failed after"));
+            let as_text = format!("{:?}", err);
+            assert!(as_text.contains("request attempt 4 failed"));
```

Comment:
this should match the max number of attempts (3) specified in the call line 112

Reply:
In practice, this count is actually the number of retries.  1 initial, 3 retries.  so, 4 attempts.